### PR TITLE
Apply Fixes suggested by Linting Tools

### LINKED ISSUE #184: Make Changes Suggested by Linting Tools
Following on from #165, the linting tools implemented should actually be used. It seems there's a _lot_ of issues that the linting tools have picked up, so there should be a separate PR for that stuff, hence the separate issue :)

### PR DESCRIPTION
**Pull request by MRichards99**
This PR will close #184. 

Flake8 gave over 400 lines of output so the major of this PR is fixing those issues. I've grouped the commits by each type of issues (e.g. missing trailing commas) so hopefully that makes things easier to follow.

I've changed one of the nox sessions from `format` to `black` due to Python builtins - so `nox -s black` is what you need now! I've also updated a couple of dependencies that had security vulnerabilities spotted by `safety` - updated `pyproject.toml` and `poetry.lock` have been committed. 

I do have the original output from flake8, so let me know if you'd like it.

### CODE REVIEW

**Review by VKTB — CHANGES_REQUESTED** [review]
Nox's lint session fails as the following suggested fixes have not been applied.

```
datagateway_api\src\main.py:12:1: F401 'datagateway_api.common.exceptions.ApiError' imported but unused
datagateway_api\src\swagger\initialise_spec.py:67:42: C812 missing trailing comma
datagateway_api\src\swagger\initialise_spec.py:68:38: C812 missing trailing comma
datagateway_api\src\swagger\initialise_spec.py:108:42: C812 missing trailing comma
datagateway_api\src\swagger\initialise_spec.py:109:38: C812 missing trailing comma
datagateway_api\src\swagger\initialise_spec.py:120:42: C812 missing trailing comma
datagateway_api\src\swagger\initialise_spec.py:121:38: C812 missing trailing comma
```

File: datagateway_api/src/resources/non_entities/sessions_endpoints.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def post(self):
                       description: Session ID
                       example: xxxxxx-yyyyyyy-zzzzzz
           400:
-            description: Bad request. User credentials were not provided in request body.
+            description: Bad request. User credentials not provided in request body
           403:
              description: Forbidden. User credentials were invalid
```
[the remark this answers is not in the document]
Comment (MRichards99) [reply datagateway_api/src/resources/non_entities/sessions_endpoints.py]: Good spot! I've changed them (searched through the other response descriptions and couldn't find any others) and also committed a rebuilt `openapi.yaml` file. I see this as resolved.

**Comment by MRichards99** [comment]
@VKTB re your commenting regarding linting output:

I've fixed those issues - I'm guessing they were caused by me merging branches after I created this PR. All done now though, and the lint session passes.

**Review by VKTB — APPROVED** [review]
Looks good

**Pull request merged**